framework/filters/AccessRule.php

```diff
@@ -161,8 +191,13 @@ protected function matchRole($user)
                 if (!$user->getIsGuest()) {
                     return true;
                 }
-            } elseif ($user->can($role)) {
-                return true;
+            } else {
+                if (!isset($roleParams)) {
```

Comment:
the problem seems to run deeper into the rbac implementation itself so after this gets merged I will create an issue about the use cases i mentioned.